### ISSUE TITLE
PDB file processing

### ISSUE THREAD

**Issue by fredrikszm** [deepchem/deepchem#1610]
Hi, I just want to ask what kind of processing was done on the PDB files in Deepchem's PDBBind dataset. I noticed that when compared to the PDB files downloaded from RCSB, they seem to be altered a lot. I know that PDBFixer allows some standardized way to clean PDB files, but I noticed the total atom count is often different as well.

For ex: 2zx6 has 7,344 total atoms on RCSB, but the 2zx6 protein pdb file downloaded from the Deepchem cloud (via load_pdbbind()) has 47,187 atoms. This is a really big difference and it's not clear to me why.

Any additional information on how Deepchem deals with and prepares PDB files for model training and prediction would be greatly appreciated!

**Comment by rbharath** [comment]
I believe that the only alterations that were done were through PDBFixer, but I'm not entirely sure. The case you point out with 2zx6 is strange and I'm not sure what's going on here.

@peastman @VIGS25 do either of you have any thoughts what might be going on here?

**Comment by VIGS25** [comment]
On the RCSB website, it says that 7344 is the number of non-hydrogen atoms. 47, 187 could be all the atoms, but the number looks still very high. 

I will check this out and get back.

**Comment by fredrikszm** [comment]
When I used Chem.MolFromPDBFile() from rdkit (which is what load_molecule in rdkit_util calls) on 2zx6_protein.pdb and check the number of atoms, I get 44319 atoms. Then, when I run add_hydrogens_to_mol, the resulting number of atoms is 47187. So the PDB file has 44319 non hydrogen atoms

**Comment by peastman** [comment]
The PDB file is at https://files.rcsb.org/view/2ZX6.pdb.  It only contains 7817 atoms.  Even adding hydrogens should only give you 15,000-20,000 atoms.  Where is RDKit getting all the others from?

**Comment by VIGS25** [comment]
I am not sure how you got the 47k number. 

I downloaded the pdb file from the DeepChem bucket, and the file from RCSB, and counted the number of non-H atoms: 

2ZX6 (PDBBIND): 23289 atoms
2ZX6 (RCSB): 7817 atoms (as @peastman said)

That said, when I run diff 2zxc_pdbbind.pdb 2zxc_rcsb.pdb, some values are output that I have added here.
[out.txt](https://github.com/deepchem/deepchem/files/3242505/out.txt)

**Comment by fredrikszm** [comment]
I also downloaded the pdb file from the DeepChem bucket and counted the atoms (lines beginning with ATOM):

[2zx6_protein.pdb.txt](https://github.com/deepchem/deepchem/files/3242928/2zx6_protein.pdb.txt)

# non-H atoms: 21893
# H atoms: 21030
total # atoms: 42923

still doesn't match the number of atoms given by the load_molecule function in rdkit_utils though

**Comment by VIGS25** [comment]
This is what I did to count,
```
mol1 = Chem.MolFromPDBFile(pddbind_file, removeHs=True)
mol2 = Chem.MolFromPDBFile(rcsb_file, removeHs=True)

print(mol1.GetNumAtoms(), mol2.GetNumAtoms())
```

@rbharath, @peastman: I checked `load_molecule` (referenced below) https://github.com/deepchem/deepchem/blob/87410a5e475a043bf68e4f16e57ba8305356153a/deepchem/utils/rdkit_util.py#L94

It looks like `load_molecule` does not remove the Hydrogens first, but then adds it later by default too. I'm not too sure I understand this idea and wonder if this is causing a discrepancy. Maybe you guys have a better idea of the reasoning to do so.

**Comment by fredrikszm** [comment]
@VIGS25 when I run 
mol1 = Chem.MolFromPDBFile(pddbind_file, **sanitize=True**, removeHs=True)
mol2 = Chem.MolFromPDBFile(rcsb_file, **sanitize=True**, removeHs=True)
print(mol1.GetNumAtoms(), mol2.GetNumAtoms())

I get what you got earlier:
2ZX6 (PDBBIND): 23289 atoms
2ZX6 (RCSB): 7817 atoms

When I run it all again with sanitize=False, I get:
2ZX6 (PDBBIND): 44319 atoms
2ZX6 (RCSB): 7817 atoms

**Comment by VIGS25** [comment]
Doesn't sanitize=True make more sense than False?

**Comment by fredrikszm** [comment]
Yes it does in practice. But I guess what I'm not understanding is why setting sanitize=False causes such an increase in number of atoms (23289 vs 44319). And also, why does adding hydrogens cause the number of atoms to increase from 23289 to 47187?

**Comment by fredrikszm** [comment]
Also @rbharath, do you know if all the alterations available from PDBFixer were used or if just some of them were (or if it varied depending on the molecule)?

**Comment by VIGS25** [comment]
For sanitize=False, I am not sure why this happens.
For the hydrogen addition, if you see, for both the pdb files, the increase is in number of atoms is almost double. As I suggested before, I think the PDBBind file is a trimer while the RCSB file is a monomer. So each monomer experiences an increase by 7000 odd hydrogens.
